src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs

```diff
@@ -68,24 +72,185 @@ internal class ResourceOperation : TypeProvider
+
+            return null;
+        }
+
+        private ResourceListMethod? GetTenantListMethod()
+        {
+            var listMethods = GetListMethods(false, true);
+            foreach (var listMethod in listMethods)
+            {
+                var restClientMethod = GetRestClientMethod(listMethod);
+                var pathSegments = restClientMethod?.Request.PathSegments;
+
+                // tenant scope
+                if (!pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/subscriptions/") &&
+                    !pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/resourceGroups/"))
+                {
+                    return listMethod;
+                }
+            }
+
+            return null;
+        }
+
+        public RestClientMethod? GetRestClientMethod(ResourceListMethod? listMethod)
+        {
+            if (listMethod != null)
+            {
+                if (listMethod.PagingMethod != null)
+                {
+                    return listMethod.PagingMethod.Method;
+                }
+                else if (listMethod.ClientMethod != null)
+                {
+                    return listMethod.ClientMethod.RestClientMethod;
+                }
+            }
+
+            return null;
+        }
+
+        protected IEnumerable<ResourceListMethod> GetListMethods(bool parentExistsInPathParam, bool returnTypeIsResourceData)
+        {
+            List<ResourceListMethod> listMethods = new List<ResourceListMethod>();
+            var parentResourceType = OperationGroup.ParentResourceType(_context.Configuration.MgmtConfiguration);
+
+            foreach (var pagingMethod in PagingMethods)
+            {
+                if (IsValidListMethod(parentExistsInPathParam, returnTypeIsResourceData, pagingMethod.Method))
+                {
+                    listMethods.Add(new ResourceListMethod(pagingMethod, null));
+                }
+            }
+
+            foreach (var method in Methods)
+            {
+                if (IsValidListMethod(parentExistsInPathParam, returnTypeIsResourceData, method.RestClientMethod))
+                {
+                    listMethods.Add(new ResourceListMethod(null, method));
+                }
+            }
+
+            return listMethods;
+        }
+
+        private bool IsValidListMethod(bool parentExistsInPathParam, bool returnTypeIsResourceData, RestClientMethod clientMethod)
+        {
+            var parentResourceType = OperationGroup.ParentResourceType(_context.Configuration.MgmtConfiguration);
+            bool isParentExistsInPathParams = false;
+            bool isReturnTypeResourceData = false;
+
+            var parentParamName = "";
+            if (clientMethod.Operation?.Requests.FirstOrDefault().Protocol.Http is HttpRequest httpRequest)
+            {
+                var path = httpRequest.Path;
+                var parentIndex = path.IndexOf(parentResourceType);
```

Comment:
I have a question: what will this `httpRequest.Path` return? For instance, if a request url is `/subscriptions/{subsID}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks/{diskName}`, is this `httpRequest.Path` returning `/subscriptions/resourceGroups/providers/Microsoft.Compute/disks`?
If not, we might never find the parentResourceType in it.
Also we might want to find this in a case insensitive way - some API has casing problems (for instance `resourceGroups` -> `resourcegroups`

Reply:
Fixed this issue, now [this](https://github.com/Azure/autorest.csharp/blob/33a18da16c450c6204a16b262f58f229ea857227/src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs#L261) checks parent in `/subscriptions/resourceGroups/providers/Microsoft.Compute/disks` path.

Reply:
[Here](https://github.com/Azure/autorest.csharp/blob/33a18da16c450c6204a16b262f58f229ea857227/test/TestProjects/MgmtMultipleParentResource/mgmtMultipleParentResource.json#L1412) is the example for this and [here](https://github.com/Azure/autorest.csharp/blob/33a18da16c450c6204a16b262f58f229ea857227/test/TestProjects/MgmtMultipleParentResource/Generated/ChildBodySubParentContainer.cs#L326) is the generated code.